[16.0] Connector typesense

sebastienbeau opened the pull request:
This PR is a refactor of this previous PR
- https://github.com/OCA/search-engine/pull/168 
- https://github.com/OCA/search-engine/pull/203

Depends on 
- [x] test refactor here: https://github.com/OCA/search-engine/pull/211

@MrTango @kobros-tech the change are ok for you ?

MrTango committed:
[ADD] connector_typesense

kobros-tech committed:
connector_typesense: Add test, improve code, clean readme

sebastienbeau commented on connector_typesense/__manifest__.py:
```diff
@@ -0,0 +1,20 @@
+    "license": "AGPL-3",
+    "author": "Derico, Kencove, Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/search-engine",
+    "maintainers": ["Kencove"],
```
@MrTango I add you in the maintainers ?

sebastienbeau commented on connector_typesense/tools/adapter.py:
```diff
@@ -0,0 +1,250 @@
+        if success_count != len(records):
+            self.clear()
+            self.index(records)
+
```
to myself: I need to check that, it's seem not to be a good idea to call a clear here.
If something wrong happen when indexing record it will purge the full index

sebastienbeau replied:
@kobros-tech can you can me some explanation why you have done this clear and index ?
I do not want to miss something.

kobros-tech replied:
as typesense can make the index (table) dynamic 

if I have company_id as the name of the company 

then I update it to object having name and value and some other fields if I want

then I must update the index, but this specific case need reindex as object fields in TS can be updated fine but can not change the type to sting without reindex

sebastienbeau replied:
Ok so it's the same with elastic. In that case we can not solve it magically (clearing the index in production is super dangerous, as you drop all your data and so shut you website). We just have to raise an error.
I am going to refactor it

kobros-tech commented:
ok

kobros-tech commented:
I think we can delete all cassette files and update the test cases with new cassette file based on the changes you made?

sebastienbeau committed:
connector_typesense: refactor code

Implement a simple update of setting (I recommend to use typesense-dashbord for advanced config)
Disable reindex and remove alias logic as it's complex and not needed, schema can be updated
Reuse generic and add some specific typesense test

sebastienbeau commented:
It's ready.
After merging it, I will create the branch 18.0 and migration the modules !

MrTango reviewed (approved):
LGTM

kobros-tech commented:
@sebastienbeau 

Are we ready to go ahead?

HviorForgeFlow commented:
could you squash fixup commits? thanks!

kobros-tech commented:
ping @sebastienbeau

sebastienbeau commented:
This PR have been merged : https://github.com/OCA/search-engine/pull/211
I am going to rebase and quash and merge this PR

sebastienbeau commented:
/ocabot merge nobump
